tests/sidecar_test.go

```diff
@@ -0,0 +1,313 @@
+		})
+	if err != nil {
+		t.Fatalf("Failed to update ", container.Name, " in group ", grp, "with new image pull policy")
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	// Update side car image with new pull policy
+	container.ImagePullPolicy = v1.PullPolicy("Always")
+	deployment, err = updateDeployment(c, depl.GetName(), ns,
+		func(depl *api.DeploymentSpec) {
+			depl.Coordinators.Sidecars = spec.GroupSideCars(grp)
+		})
+	if err != nil {
+		t.Fatalf("Failed to update ", container.Name, " in group ", grp, "with new image pull policy")
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	// Remove all sidecars again
+	spec.ClearGroupSideCars(grp)
+	deployment, err = updateDeployment(c, depl.GetName(), ns,
+		func(depl *api.DeploymentSpec) {
+			depl.Coordinators.Sidecars = spec.GroupSideCars(grp)
+		})
+	if err != nil {
+		t.Fatalf("Failed to remove all sidecars from group ", grp)
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	// Adding containers to coordinators and db servers
+	spec.AddSideCar(grp, v1.Container{Image: "busybox", Name: "busybox", Command: ["sh", "-c", "sleep 1000"]})
+	spec.AddSideCar(dbs, v1.Container{Image: "busybox", Name: "busybox", Command: ["sh", "-c", "sleep 1000"]}})
+	deployment, err = updateDeployment(c, depl.GetName(), ns,
+	func(depl *api.DeploymentSpec) {
+		depl.Coordinators.Sidecars = spec.GroupSideCars(grp)
+	})
+	if err != nil {
+		t.Fatalf("Failed to add a container to both coordinators and db servers", grp)
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	// Clear containers from both groups
+	spec.ClearGroupSideCars(grp)
+	spec.ClearGroupSideCars(dbs)
+	deployment, err = updateDeployment(c, depl.GetName(), ns,
+	func(depl *api.DeploymentSpec) {
+		depl.Coordinators.Sidecars = spec.GroupSideCars(grp)
+	})
+	if err != nil {
+		t.Fatalf("Failed to delete all containers from both coordinators and db servers", grp)
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	// Adding containers to coordinators again
+	spec.AddSideCar(grp, v1.Container{Image: "busybox", Name: "busybox", Command: ["sh", "-c", "sleep 1000"]})
+	deployment, err = updateDeployment(c, depl.GetName(), ns,
+	func(depl *api.DeploymentSpec) {
+		depl.Coordinators.Sidecars = spec.GroupSideCars(grp)
+	})
+	if err != nil {
+		t.Fatalf("Failed to add a container to both coordinators and db servers", grp)
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	// Clear containers from coordinators and add to db servers
+	spec.ClearGroupSideCars(grp)
+	spec.AddSideCar(dbs, v1.Container{Image: "busybox", Name: "busybox", Command: ["sh", "-c", "sleep 1000"]}})
+	deployment, err = updateDeployment(c, depl.GetName(), ns,
+	func(depl *api.DeploymentSpec) {
+		depl.Coordinators.Sidecars = spec.GroupSideCars(grp)
+	})
+	if err != nil {
+		t.Fatalf("Failed to delete all containers from both coordinators and db servers", grp)
+	} else {
+		t.Log("Updated deployment")
+	}
+
+	removeDeployment(c, depl.GetName(), ns)
+
```

Comment:
Just curious: what are our whitespace style guide rules

Reply:
we use `go fmt`.